main.js

```diff
@@ -35,21 +35,19 @@ function runOpenAndClose(property) {
   mobileMenu.style.display = property;
 }
 
-async function exampel() {
+async function topThree() {
   const url = 'https://lernia-sjj-assignments.vercel.app/api/challenges';
   const response = await fetch(url);
   const data = await response.json();
+  const sortedData = JSON.parse(JSON.stringify(data.challenges)).sort((x, y) => y.rating - x.rating);
```

Comment:
I'm guessing what you're trying to do here (with `JSON.parse` and `JSON.stringify`) is to copy the array so that your sorting does not affect the original array. I think that's a good ambition, but not a great solution.

Copying the array before sorting is often good, because the [`Array.prototype.sort`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) function ["sorts in place"](https://en.wikipedia.org/wiki/In-place_algorithm).  There are pros and cons to this, but of course it's annoying if you want to use the array for something else.

In this case, I don't think you do use the array for anything else, so I think the copying is unnecessary. But if you do want to copy an array in the future, the best way to do that is probably `Array.prototype.concat()`. Using `concat` the code would have looked like this:

```js
const sortedData = data.challenges.concat().sort((x, y) => y.rating - x.rating);
```